### PR TITLE
Update ci.yml

### PR DESCRIPTION
**Pull request by PieterTack**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -199,29 +180,12 @@ jobs:
         brew uninstall -f bazel bazelisk
         brew update
         brew upgrade || true
-        brew install hdf5 git autoconf automake libtool curl wget pkg-config gsl numpy meson ninja
+        brew install hdf5 git libtool curl wget pkg-config gsl numpy meson ninja
```
Comment (PieterTack) [inline .github/workflows/ci.yml]: I suppose this is ok? Autoconf and automake not needed anymore? What about libtool, what is that used for?

Comment (tschoonj) [reply .github/workflows/ci.yml]: You can also safely get rid of `libtool`, which is also part of the GNU autotools

**Comment by PieterTack** [comment]
So as far as I understood, meson never uses Automake.am files, so I went through all directories and removed them. Also configure.ac shouldn't be used by meson, so got rid of that.
Then I made some changes to the ci.yml... There are a few lines of code that I'm not sure about.... I added some comments there.

**Comment by PieterTack** [comment]
In 'ci.yml', in build msys2/setup-msys2@v2; is it safe to remove make etc here as well? I guess not, as it is then used to make xraylib etc? Or can we do this differently on msys2 as well?

**Comment by PieterTack** [comment]
Status note:
- visual studio: "found pkg-config 'C:\\Strawberry\\perl\\bin\\pkg-config.BAT' but it is Strawberry Perl and thus broken. Ignoring..."
--> how does one properly install pkg-config on windows system/visualstudio?

- ubuntu:latest: " OpenMP found but omp.h missing." --> somehow meson setup does not find omp.h... From what I find I should add the -fopenmp flag during compilation, but as this isn't the compilation step yet I'm not understanding the problem at hand. Probably a path is missing somewhere? But why, as I think before this all this wasn't an issue for ubuntu systems?

**Comment by tschoonj** [comment]
@PieterTack I got the visual studio build working after a lot of trial and error. Would be great if you could review if you have a moment.

**Comment by tschoonj** [comment]
Note that this PR already includes the changes necessary to support HDF5 1.14, as this version is in use by both msys2 and vcpkg.

**Review by tschoonj — APPROVED** [review]

**Pull request merged**